dotCMS/src/main/java/com/dotcms/contenttype/business/RelationshipFactoryImpl.java

```diff
@@ -135,125 +136,140 @@ public  Relationship byTypeValue(final String typeValue) {
-            + "relationship.parent_structure_inode = parentstruct.inode and "
-            + "relationship.child_structure_inode = childstruct.inode and (relationship.parent_structure_inode = '"
-            + type.id() + "' or relationship.child_structure_inode = '" + type.id() + "')  order by " + orderBy;
-      }
-      try {
-        HibernateUtil dh = new HibernateUtil(Relationship.class);
-        dh.setSQLQuery(query);
-        list = dh.list();
-      } catch (DotHibernateException e) {
-        Logger.error(OldRelationshipFactory.class, e.getMessage(), e);
-      }
-      return list;
-    }
-    
-    
-    
-    
-    
-    
-    
-    @Override
-    public  List<Contentlet> dbRelatedContent(Relationship relationship, Contentlet contentlet) throws DotDataException {
-        String stInode = contentlet.getStructure().getInode();
+    public  List<Contentlet> dbRelatedContent(final Relationship relationship, final Contentlet contentlet) throws DotDataException {
+        final String stInode = contentlet.getContentTypeId();
 
-        boolean selfJoinRelationship = relationship.getParentStructureInode().equalsIgnoreCase(stInode)
+        final boolean selfJoinRelationship = relationship.getParentStructureInode().equalsIgnoreCase(stInode)
             && relationship.getChildStructureInode().equalsIgnoreCase(stInode);
 
-        boolean hasParent = !selfJoinRelationship && relationship.getParentStructureInode().equalsIgnoreCase(stInode);
+        final boolean hasParent = !selfJoinRelationship && relationship.getParentStructureInode().equalsIgnoreCase(stInode);
 
         return dbRelatedContent(relationship, contentlet, hasParent);
     }
 
     @Override
-    public  List<Contentlet> dbRelatedContent(Relationship relationship, Contentlet contentlet,
-            boolean hasParent) throws  DotDataException {
+    public  List<Contentlet> dbRelatedContent(final Relationship relationship, final Contentlet contentlet,
+            final boolean hasParent) throws  DotDataException {
         return dbRelatedContent (relationship, contentlet, hasParent, false, "tree_order");
     }
+
+    @SuppressWarnings("deprecation")
+    public  List<Contentlet> dbRelatedContent(final Relationship relationship, final Contentlet contentlet,
+            final boolean hasParent, final boolean live, final String orderBy) throws  DotDataException {
+        List<Contentlet> matches = new ArrayList<Contentlet>();
+
+        if(contentlet == null || !InodeUtils.isSet(contentlet.getIdentifier()))
+            return matches;
+        String iden = "";
+        try{
+            iden = APILocator.getIdentifierAPI().find(contentlet).getInode();
+        }catch(DotHibernateException dhe){
+            Logger.error(this.getClass(), "Unable to retrieve Identifier", dhe);
+        }
+
+        if(!InodeUtils.isSet(iden))
+            return matches;
+
+        if (hasParent) {
+            if (live)
```

Comment:
this could be
```
matches = live? dbRelatedContentByParent(iden, relationship.getRelationTypeValue(),true, orderBy):
dbRelatedContentByParent(iden, relationship.getRelationTypeValue(),false, orderBy);
```